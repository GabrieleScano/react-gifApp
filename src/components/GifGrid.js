import React from 'react'
import PropTypes from 'prop-types'  

import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import { Spinner } from './Spinner'

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{category}</h3>

      {loading && <Spinner />}

      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

