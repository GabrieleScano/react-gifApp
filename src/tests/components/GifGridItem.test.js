import React from 'react'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Test <GifGridItem /> component', () => {
  const title = 'Title'
  const url = 'https://localhost/algo.jpg'
  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test('should show the component', () => {
    expect(wrapper).toMatchSnapshot()
  });

  test('should have a p with the title', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  });
  test('the image should have the props url and alt', () => {
    const img = wrapper.find('img')
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  });
  test('should have class animate__fadeIn', () => {
    const div = wrapper.find('div')
    const className = div.prop('className')
    expect(className.includes('animate__fadeIn')).toBe(true)
  });
});
