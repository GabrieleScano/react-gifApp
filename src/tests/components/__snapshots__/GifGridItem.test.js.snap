// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> component should show the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://localhost/algo.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
